src/lib/utils/debounce.js

```diff
@@ -1,15 +1,18 @@
 /**
 	A simple debounce function taken from here https://www.freecodecamp.org/news/javascript-debounce-example/
-	@param {function} func The function to debounce.
+	@param {Function} func The function to debounce.
 	@param {number} timeout The time in ms to wait.
-	@returns {function}
+	@returns {Function}
 */
-export default function debounce(func, timeout = 300) {
+const debounce = (func, timeout = 300) => {
```

Comment:
Changing this to an arrow function was the solution I found to `this` not being defined.

Reply:
Hm, in an arrow function, the reference of `this` changes. I imagine it still works because `this` isn't actually relevant for this call?

Reply:
Hm, I'm not sure I fully understand the "An outer value of 'this' is shadowed by this container." error message that was there. 

Another approach might be to use something like here: https://decipher.dev/30-seconds-of-typescript/docs/debounce/

With JSDoc:

```javascript
export default function debounce(func, timeout = 300) {
	/** @type {string | number | NodeJS.Timeout | undefined} */
	let timer;
	return function (/** @type {any} */ this, /** @type {any[]} */...args) {
		clearTimeout(timer);
		timer = setTimeout(() => {
			func.apply(this, args);
		}, timeout);
	};
}
```

Reply:
Got it – I imagine that error is saying that because `this` exists in the parent scope, any child `this` is shadowed by it. I have a feeling it's not a big deal. I'll take a look at this other function, too, for good measure.

Sorry this has taken me longer than usual to review. I am on it, though, and really appreciate the work you've put into it!

Reply:
Absolutely no worries about reviews taking long! None of this is urgent and I appreciate your careful reviews.

In any case, might also be an option to silence that warning message. I think the main benefit of having a cleanly linted version is to easily see that refactoring or new features don't introduce new problems.

Reply:
Totally agree on that point about the value it brings.